tap_linkedin_ads/streams.py

```diff
@@ -0,0 +1,707 @@
+                                catalog=catalog,
+                                last_datetime=child_obj.get_bookmark(state, start_date),
+                                date_window_size=date_window_size,
+                                parent_id=parent_id)
+                        else:
+                            child_total_records, child_batch_bookmark_value = child_obj.sync_endpoint(
+                                client=client,
+                                catalog=catalog,
+                                state=state,
+                                page_size=page_size,
+                                start_date=start_date,
+                                selected_streams=selected_streams,
+                                date_window_size=date_window_size,
+                                parent_id=parent_id)
+
+                        child_batch_bookmark_dttm = strptime_to_utc(child_batch_bookmark_value)
+                        child_max_bookmark = child_max_bookmarks.get(child_stream_name)
+                        child_max_bookmark_dttm = strptime_to_utc(child_max_bookmark)
+                        if child_batch_bookmark_dttm > child_max_bookmark_dttm:
+                            # Update bookmark for child stream.
+                            child_max_bookmarks[child_stream_name] = strftime(child_batch_bookmark_dttm)
+
+                        LOGGER.info('Synced: %s, parent_id: %s, total_records: %s',
+                                    child_stream_name,
+                                    parent_id,
+                                    child_total_records)
+                        LOGGER.info('FINISHED Syncing: %s', child_stream_name)
+
+            # Pagination: Get next_url
+            next_url = None
+            links = data.get('paging', {}).get('links', [])
+            for link in links:
+                rel = link.get('rel')
+                if rel == 'next':
+                    # Get URL to fetch next page
+                    href = link.get('href')
+                    if href:
+                        next_url = 'https://api.linkedin.com{}'.format(urllib.parse.unquote(href))
+
+            if self.tap_stream_id in selected_streams:
+                LOGGER.info('%s: Synced page %s, this page: %s. Total records processed: %s',
+                            self.tap_stream_id,
+                            page,
+                            record_count,
+                            total_records)
+            page = page + 1
+
+        # Write child stream's bookmarks
+        for key, val in list(child_max_bookmarks.items()):
+            write_bookmark(state, val, key)
+
+        return total_records, max_bookmark_value
+
+    # pylint: disable=too-many-branches,too-many-statements,unused-argument
+    def sync_ad_analytics(self, client, catalog, last_datetime, date_window_size, parent_id=None):
+        """
+        Sync method for ad_analytics_by_campaign, ad_analytics_by_creative
```

Comment:
If this function is specific to 2 streams only, then create one general parent class for it and keep the above function within it.

Reply:
I can not see any benefit to creating a separate class because in the separate class also, we require to define the sync method in a separate class.